### PR TITLE
[server] add unified routes to serve records and trigger actions

### PR DESCRIPTION
**Pull request by TBonnin**
fetching records  and triggering actions can now be done via a `METHOD /v1/:path` endpoint. Example: `GET /v1/hr/users`or `POST /v1/finance/payments`: 
```
curl --request GET -v \ 
  --url http://localhost:3003/v1/hr/users \
  --header 'Authorization: Bearer xxx' \
  --header 'Connection-Id: connection-2' \
  --header 'Content-Type: application/json' \
  --header 'Provider-Config-Key: github'

curl --request POST -v \ 
  --url http://localhost:3003/v1/finance/payments \
  --header 'Authorization: Bearer xxx' \
  --header 'Connection-Id: connection-2' \
  --header 'Content-Type: application/json' \
  --header 'Provider-Config-Key: github' \
∙ --data { ... }
```

The endpoints are defined in the nango.yaml file as follow.
There could be multiple `endpoints` values that would each correspond to a `returns` value
```
// nango.yaml
integrations:
  github:
    github-issue-example:
      type: sync
      runs: every half hour
      returns:
        - GithubIssue
      endpoints:
        - GET: /engineering/issues
```

### CODE REVIEW

**Comment by bastienbeurier** [comment]
I can't believe how minimal this change was 😄 Nicely done!

**Comment by khaliqgant** [comment]
Nice! Is this case insensitive? Or is it only that GithubIssue will work, not githubissue?

**Comment by bastienbeurier** [comment]
> Nice! Is this case insensitive? Or is it only that GithubIssue will work, not githubissue?

I agree that it might remove some confusion to convert paths to lowercase. I don't recall seing URL paths capitalized. Maybe kebab case? `GithubIssues` becomes `github-issues`?

**Comment by TBonnin** [comment]
It was indeed case sensitive since the model name is directly passed as is in the sql query. Here is the commit to support kebab-case like `/api/github-issue` https://github.com/NangoHQ/nango/pull/1188/commits/6cbfc2817e346a6e48787ed215e9a84508f427ec

**Comment by TBonnin** [comment]
I just realised the  webapp routes already uses the endpoint prefix `/api/`.
What about using `/model/:model` for this route? @khaliqgant @bastienbeurier

File: packages/server/lib/controllers/sync.controller.ts [hunk 1]
```diff
@@ -228,13 +241,70 @@ class SyncController {
...
+            const {
+                success,
+                error,
+                response: connection
+            } = await connectionService.getConnection(connectionId as string, providerConfigKey as string, environmentId);
+
+            if (!success) {
+                errorManager.errResFromNangoErr(res, error);
+                return;
+            }
```
Comment (TBonnin) [inline packages/server/lib/controllers/sync.controller.ts:271]: There is a bit of duplication with the request validation. We can refactor it when we get rid of the old `sync/records` and `trigger/action` routes.

**Comment by khaliqgant** [comment]
Closing in favor of https://github.com/NangoHQ/nango/pull/1266. Thanks Thomas!

**Pull request closed without merging**